addons/survey/models/survey.py

```diff
@@ -0,0 +1,1159 @@
+    # The answers !
+    user_input_line_ids = fields.One2many('survey.user_input_line',
+                                           'user_input_id', string='Answers', copy=True)
+
+    # URLs used to display the answers
+    result_url = fields.Char(related='survey_id.result_url',
+                                 string="Public link to the survey results")
+    print_url = fields.Char(related='survey_id.print_url',
+                                string="Public link to the empty survey")
+
+    quizz_score = fields.Float(compute="_compute_quizz_score", string="Score for the quiz")
+
+    _sql_constraints = [
+        ('unique_token', 'UNIQUE (token)', 'A token must be unique!'),
+        ('deadline_in_the_past', 'CHECK (deadline >= date_create)', 'The deadline cannot be in the past')
+    ]
+
+    @api.multi
+    def _compute_quizz_score(self):
+        for user_input in self:
+            user_input.quizz_score = sum([uil.quizz_mark for uil in user_input.user_input_line_ids])
+
+    @api.multi
+    def action_survey_resent(self):
+        ''' Sent again the invitation '''
+        self.ensure_one()
+        context = dict(self.env.context or {})
+        context.update({
+            'survey_resent_token': True,
+            'default_partner_ids': self.partner_id.id and [self.partner_id.id] or [],
+            'default_multi_email': self.email or "",
+            'default_public': 'email_private',
+        })
+        return self.survey_id.env['survey.survey'].with_context(context).action_send_survey()
+
+    @api.multi
+    def action_view_answers(self):
+        ''' Open the website page with the survey form '''
+        self.ensure_one()
+        user_input = self.read(['print_url', 'token'])[0]
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "View Answers",
+            'target': 'self',
+            'url': '%s/%s' % (user_input['print_url'], user_input['token'])
+        }
+
+    @api.multi
+    def action_survey_results(self):
+        ''' Open the website page with the survey results '''
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "Survey Results",
+            'target': 'self',
+            'url': self.read(['result_url'])[0]['result_url']
+        }
+
+    @api.one
+    def copy_data(self, default=None):
+        raise UserError(_('You cannot duplicate this element!'))
+
+    @api.model
+    def do_clean_emptys(self, automatic=False):
+        ''' Remove empty user inputs that have been created manually
+            (used as a cronjob declared in data/survey_cron.xml) '''
+        empty_user_inputs = self.search([('input_type', '=', 'manually'),
+                                                     ('state', '=', 'new'),
+                                                     ('date_create', '<', fields.Datetime.to_string((datetime.datetime.now() - datetime.timedelta(hours=1))))])
+        if empty_user_inputs:
```

Comment:
recordset is smart now. so no need to check existence or not. just call empty_user_inputs.unlink()